scripts/lib/sphinx/addFrontMatter.test.ts

```diff
@@ -0,0 +1,79 @@
+// This code is a Qiskit project.
+//
+// (C) Copyright IBM 2024.
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+import { expect, test } from "@jest/globals";
+
+import addFrontMatter from "./addFrontMatter";
+import { Pkg } from "../sharedTypes";
+import { SphinxToMdResult } from "./SphinxToMdResult";
+
+test("addFrontMatter()", () => {
+  const results: SphinxToMdResult[] = [
+    {
+      markdown: "# Hardcoded!",
+      meta: {
+        hardcoded_frontmatter: "title: hardcoded\ndescription: Hello world!",
+      },
+      images: [],
+      isReleaseNotes: false,
+    },
+    {
+      markdown: "# Python API",
+      meta: {
+        python_api_name: "quantum_software.MyClass",
+        python_api_type: "class",
+      },
+      images: [],
+      isReleaseNotes: false,
+    },
+    {
+      markdown: "# Some release notes!",
+      meta: {},
+      images: [],
+      isReleaseNotes: true,
+    },
+  ];
+  const pkg = {
+    versionWithoutPatch: "0.0",
+    hasSeparateReleaseNotes: true,
+    title: "My Quantum Software Project",
+  } as Pkg;
```

Comment:
This is an important trick to make tests more readable, rather than us having to give dummy values for attributes that aren't relevant. We should audit existing tests to use this when possible.